config/default/contact-summary-extras.js

```diff
@@ -0,0 +1,445 @@
+  //If pregnancy registration and no LMP, make it available at registration and until 280 days + 6 weeks from the date of registration. 
+  //If no pregnancy registration, previous delivery date should be at least 7 months ago.
+  if (thisContact.type !== 'person') return false;
+  const latestPregnancy = getNewestReport(allReports, pregnancyForms);
+  const latestDelivery = getNewestReport(allReports, deliveryForms);
+  if (!latestPregnancy && !latestDelivery) {
+    return true;//no pregnancy, no previous delivery
+  }
+  else if (latestDelivery && (!latestPregnancy || latestDelivery.reported_date > latestPregnancy.reported_date)) {
+    //no pregnancy registration, previous delivery date should be at least 7 months ago.
+    return getDeliveryDate(latestDelivery) < today.clone().subtract(7, 'months');
+  }
+
+  else if (latestPregnancy) {//pregnancy registration
+    if (isPregnancyForm(latestPregnancy)) {
+      const lmpDate = getMostRecentLMPDateForPregnancy(allReports, latestPregnancy);
+      if (!lmpDate) {//no LMP, show until 280 days + 6 weeks from the date of registration
+        return moment(latestPregnancy.reported_date).clone().startOf('day').add(280 + 6 * 7, 'days').isSameOrBefore(today);
+      }
+      else {//Pregnancy registration with LMP
+        const edd = getMostRecentEDDForPregnancy(allReports, latestPregnancy);
+        //at least 6 months ago, no more than EDD + 6 weeks
+        return today.isBetween(lmpDate.clone().add(6, 'months'), edd.clone().add(6, 'weeks'));
+      }
+    }
+  }
+  return false;
+}
+
+function getRecentANCVisitWithEvent(allReports, pregnancyReport, event) { //miscarriage, abortion, refused, migrated
+  const followUps = getSubsequentPregnancyFollowUps(allReports, pregnancyReport);
+  const latestFollowup = getNewestReport(followUps, antenatalForms);
+  if (latestFollowup && getField(latestFollowup, 'pregnancy_summary.visit_option') === event) {
+    return latestFollowup;
+  }
+}
+
+function getSubsequentDeliveries(allReports, pregnancyReport, withinLastXDays) {
+  return allReports.filter(function (report) {
+    return (isDeliveryForm(report)) &&
+      report.reported_date > pregnancyReport.reported_date &&
+      (!withinLastXDays || report.reported_date >= (today.clone().subtract(withinLastXDays, 'days')));
+  });
+}
+
+function getSubsequentPregnancyFollowUps(allReports, pregnancyReport) {
+  let lmpDate = getLMPDateFromPregnancy(pregnancyReport);
+  if (!lmpDate) { //LMP Date is not available, use reported date
+    lmpDate = moment(pregnancyReport.reported_date);
+  }
+  const subsequentVisits = allReports.filter(function (visitReport) {
+    return isPregnancyFollowUpForm(visitReport) &&
+      visitReport.reported_date > pregnancyReport.reported_date &&
+      moment(visitReport.reported_date) < lmpDate.clone().add(MAX_DAYS_IN_PREGNANCY, 'days');
+  });
+  return subsequentVisits;
+}
+
+function countANCFacilityVisits(allReports, pregnancy) {
```

Comment:
Do you have two `countANCFacilityVisits`? One for contact-summary and one for nools? They seem identical ...   You can move this function into a new file called "common.js" or something, and `require` that single file. This will help avoid code duplication.

Reply:
They are slightly different but can be made into a common function. However, it depends on other time-related functions. I have used `momentjs` in `contact-summary-extras` but couldn't use it in `nools-extras` because of a run-time error. That's why I am having trouble using common functions across nools and contact-summary.

Reply:
Ahhh interesting. Thanks for the explanation.